---
title: "Manuscript Title"
description: brief description about manuscript
author:                                              
  - name: Kewalin Samart                             # author information
    affiliation: Mathematics, Michigan State University
    affiliation_url: https://github.com/JRaviLab
    url: https://github.com/KewalinSamart
    orcid_id: 0000-0003-3134-8846  
  - name: second author name 
    affiliation: affillation name
    affiliation_url: affillation link 
    url: link to author's webpage/social media e.g. GitHub profile
    orcid_id: 0000-000X-XXXX-XXXX
  - name: third author name 
    affiliation: affillation name
    affiliation_url: affillation link 
    url: link to author's webpage/social media e.g. GitHub profile
    orcid_id: 0000-000X-XXXX-XXXX
date: '2021-05-09'
output:                                               
  distill::distill_article:                           
    toc: true                                         # table of content
    theme: theme.css                                  # enable adding css theme
graphics: yes                                         
font-family: Helvetica                                # text font
images: data/figures                                  # path to images
bibliography: refs/connectivity_score_review.bib      # path to bibtex file
csl: refs/briefings-in-bioinformatics.csl             # path to csl file (journal format)
biblio-style: apa                                     # citation style
citation_url: https://arxiv.org/abs/2009.09317        # cite the original article (this will automatically create citation boxes after reference section) 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
```

# Header 1
## Sub Header 1
### Sub Header 2

## Normal text and in-line TeX
The basis of several recent methods for drug repurposing is the key principle that an efficacious drug will reverse the disease molecular 'signature' with minimal side-effects. This principle was defined and popularized by the influential 'connectivity map' study in 2006 regarding reversal relationships between disease- and drug-induced gene expression profiles, quantified by a disease-drug 'connectivity score.' Over the past 15 years, several studies have proposed variations in calculating connectivity scores towards improving accuracy and robustness in light of massive growth in reference drug profiles. However, these variations have been formulated inconsistently using various notations and terminologies even though they are based on a common set of conceptual and statistical ideas. Therefore, we present a systematic reconciliation of multiple disease-drug similarity metrics ($ES$, $css$, $Sum$, $Cosine$, $XSum$, $XCor$, $XSpe$, $XCos$, $EWCos$) and connectivity scores ($CS$, $RGES$, $NCS$, $WCS$, $Tau$, $CSS$, $EMUDRA$) by defining them using consistent notation and terminology. In addition to providing clarity and deeper insights, this coherent definition of connectivity scores and their relationships provides a unified scheme that newer methods can adopt, enabling the computational drug-development community to compare and investigate different approaches easily. To facilitate the continuous and transparent integration of newer methods, this article will be available as a live document (<https://jravilab.github.io/connectivity_scores>) coupled with a GitHub repository (<https://github.com/jravilab/connectivity_scores>) that any researcher can build on and push changes to.


## In-text Citation Examples

The past few decades has seen a rapid increase in computational, experimental, and clinical drug repositioning/repurposing approaches owing to the appeal of reduced costs and drug discovery time [@brown_standard_2017; @shameer_computational_2015; @parvathaneni_drug_2019]. Drug repurposing works on the principle that drugs have multiple modes of action, targets, and off-targets, that can be exploited to identify new indications [@brown_standard_2017]. This principle has been leveraged to identify novel therapeutic candidates for several diseases [@brown_standard_2017 ; @shameer_systematic_2018]. Approaches and resources for drug repurposing have been broadly summarized and discussed elsewhere [@pushpakom_drug_2019; @shameer_computational_2015]. With the accumulation of massive drug and disease data collections, computational methods and databases have now become an indispensable component of the drug repurposing workflow [@shameer_computational_2015; @tanoli_exploration_2020]. Nearly all these methods leverage high-throughput gene expression profiles abundantly available for drugs and diseases to find novel associations [@lamb_connectivity_2006; @duan_lincs_2014; @barrett_ncbi_2007]. These expression profiles can be used to derive a characteristic molecular imprint, *i.e.,* a signature, of a disease or drug perturbation in a tissue [@huang_perturbational_2019]. Large compendia of such transcriptomic signatures have been created for thousands of drugs based on the differential gene expression of various cell lines with or without drug perturbation. Computational methods then use these compendia to predict repurposed candidates for a disease either based on the (dis)similarity of a drug's expression signature to that disease's expression signature [@keenan_connectivity_2019] or based on similarity to the signatures of other drugs previously linked to the disease [@iorio_discovery_2010; @carrella_mantra_2014].

In this article, we will focus on these widely-used expression-based methods for drug repurposing collectively referred to as "drug-disease connectivity analysis" [@keenan_connectivity_2019]. A typical instance of this analysis is presented in **Figure 1** where novel drug indications for a particular disease of interest are identified based on the extent to which the ranked drug-gene signature is a "reversal" of the disease gene signature ([@dudley_computational_2011; @sirota_discovery_2011] **Fig. 1**). Connectivity-based drug repurposing has been used to discover drugs in various cancers and non-cancer diseases [@parvathaneni_drug_2019].

## Inserting Figure with Legend
```{r, simple, eval=FALSE, echo=TRUE}
{r echo=FALSE, fig.width=12, fig.height=9, out.width="100%"}
knitr::include_graphics(here::here("data/figures/fig1-disease_drug_connectivity.png"))
```

```{r echo=FALSE, fig.width=12, fig.height=9, out.width="100%"}
knitr::include_graphics(here::here("data/figures/fig1-disease_drug_connectivity.png"))
```

```{r, eval=FALSE, echo=TRUE}
<font size = "2">**Figure 1. Drug-disease connectivity. -figure legend- </font>
```
<font size = "2">**Figure 1. Drug-disease connectivity. A.** *Gene expression signatures.* Gene expression signatures. $\vec{R}$ and $\vec{S}$ are rank-ordered drug and disease gene expression signatures going from the most significantly up-regulated genes to the most significantly down-regulated genes. $S$ is the full set of genes with disease data. Without any loss in generality, only the subset of disease genes that are also part of $R$ are considered throughout (*i.e.,* $S \subseteq R$ ). $S^{+}$ and $S^{-}$ correspond to the set of most up- and down-regulated sets of disease genes, respectively. **B.** *Connectivity*. Positions of $S^{+}$ and $S^{-}$ disease genes in the ranked drug list, $\vec{R}$, determine the signs of enrichment scores ($ES$; $ES_{up}$, $ES_{down}$). Positive connectivity is defined as the case when the disease signature and drug profile show similar perturbations, *i.e.,* when $ES_{up}$ is positive and/or when $ES_{down}$ is negative. This happens when $S^{+}$ predominantly appears towards the top of the drug profile or when $S^{-}$ appears predominantly towards the bottom of the drug profile (scenarios 1 and 4). Conversely, negative connectivity is defined as the case when the disease signature and drug profile show dissimilar perturbations, *i.e.,* when $ES_{up}$ is negative and/or when $ES_{down}$ is positive. This happens when $S^{+}$ predominantly appears towards the bottom of the drug profile or when $S^{-}$ appears towards the top of the drug profile (scenarios 2 and 3). Negative connectivity indicates drug reversal of disease signature. </font>

## Manually Inserting Table
```{r, eval=FALSE, echo=TRUE}
**Table 3. CMap 1.0 Notations**

| *Current Notation*   | *Previous Notation* | *Description*                                                                    |
|----------------------|---------------------|----------------------------------------------------------------------------------|
| $CS$                 | $S^{i}$             | connectivity score; normalized connectivity score across all treatment instances |
| $t$                  | $i$                 | treatment instances                                                              |
| $cs$                 | $s^{i}$             | connectivity score for each treatment instance                                   |
| $ES$                 | $ks$                | enrichment score                                                                 |
| $r_{drg}(gs_{i})$    | $V(j)$              | position of $gs_{i}$ in $\vec{R}$                                                |
| $N_{R}$, $N_{S_{X}}$ | $t,n$               | number of genes in $\vec{R}$ and $S_{X}$                                         |
```
**Table 3. CMap 1.0 Notations**

| *Current Notation*   | *Previous Notation* | *Description*                                                                    |
|----------------------|---------------------|----------------------------------------------------------------------------------|
| $CS$                 | $S^{i}$             | connectivity score; normalized connectivity score across all treatment instances |
| $t$                  | $i$                 | treatment instances                                                              |
| $cs$                 | $s^{i}$             | connectivity score for each treatment instance                                   |
| $ES$                 | $ks$                | enrichment score                                                                 |
| $r_{drg}(gs_{i})$    | $V(j)$              | position of $gs_{i}$ in $\vec{R}$                                                |
| $N_{R}$, $N_{S_{X}}$ | $t,n$               | number of genes in $\vec{R}$ and $S_{X}$                                         |

## Writing Mathematical Equations using LaTeX Math 
```{r, eval=FALSE, echo=TRUE}
$$
\sigma = \sqrt{ \frac{1}{N} \sum_{i=1}^N (x_i -\mu)^2}
$$
```

$$
\sigma = \sqrt{ \frac{1}{N} \sum_{i=1}^N (x_i -\mu)^2}
$$

LaTeX Math: https://en.wikibooks.org/wiki/LaTeX/Mathematics